app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/provision/cloning.rb

```diff
@@ -121,11 +126,19 @@ def make_request_clone(clone_options)
   end
 
   def check_task_clone_to_template(clone_task_ref)
-    source.with_provider_connection(:service => "PCloudTasksApi") do |api|
-      task = api.pcloud_tasks_get(clone_task_ref)
-      stop = (task.status != 'capturing')
+    source.with_provider_connection(:service => 'PCloudJobsApi') do |api|
+      job = api.pcloud_cloudinstances_jobs_get(source.ext_management_system.uid_ems, clone_task_ref)
+      stop = (job.status.state == 'completed')
       phase_context[:cloud_api_completion_time] = Time.zone.now.utc if stop
-      return stop, task.status_detail
+      # Unfortunately the Power Cloud PVM instance capture API response does
+      # not include the new image ID. It does include the new image name in the
+      # status message. In order to handle cases where automate alters the name
+      # from the dialog (options[:vm_name]) we parse the name from the Job
+      # statue message and assign it to phase_context[:tempalte_name] to use as
+      # a vmdb lookup key later
+      phase_context[:template_name] = job.status.message.match(/image named (\S+?) completed/)[1] if stop
```

Comment:
Minor but prefer
```
if condition
  line 1
  line 2
end
```

Rather than
```
line1 if condition
line2 if condition
```

Reply:
`/image named (\S+?) completed/`

Assuming image names can't contain whitespace?

Reply:
Can you get the real name from `:vm_target_name`?  We have to have access to the full name not just the dialog selected name when we build the clone_options.

https://github.com/ManageIQ/manageiq-providers-ibm_cloud/blob/master/app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/provision/cloning.rb#L56

Reply:
> Assuming image names can't contain whitespace?

That's correct - no whitespace allowed in the name.

Reply:
> Can you get the real name from `:vm_target_name`? We have to have access to the full name not just the dialog selected name when we build the clone_options.

I'm not sure where automate changes the names, but I agree matching the name from the response isn't great. If `:vm_target_name` has the update name I'll refactor to use that.

Reply:
`:vm_name` from the the options hash works - just updated the patch